components/bl-list-component/src/list-content.js

```diff
@@ -0,0 +1,53 @@
+import { List } from './list';
+
+export function ListContent({
+  component, eventHandlers, child}) {
+
+  const { color, fontSize, listItems, type} = component;
+
+  const itemStyles = {
+    color   : color,
+    fontSize: fontSize
+  };
+
+  const onItemClick = (key) => {
+    eventHandlers.onClickListItem({item: key});
+  }
+
+ const getItems = (items) => {
```

Comment:
That was named a Render Function. In general, we wan't create them - because we write on react and there are preferable component approach.

You made it almost right - but there is a mistake in entities.

We must have an two components:

ListItem - that render an ordinary item (li) and creates a sibling List if it has children.
List - that render a collection of items

Reply:
Also, pseudo code for that looks like:

```
function List({ children }) {
  return (
    <ul>
      { children.map(child => {
        return (
          <Item { ...child }/>
        )
      }) }
    </ul>
  )
}

function Item(itemProps) {
  return (
    <>
      <li>{ itemProps.text } </li>
      { itemProps.children && <List children={ itemProps.children }/> }
    </>
  )
}
```